pkg/conversion/deep_equal.go

```diff
@@ -174,6 +174,9 @@ func (e Equalities) deepValueEqual(v1, v2 reflect.Value, visited map[visit]bool,
 
 	switch v1.Kind() {
 	case reflect.Array:
+		if v1.Len() != v2.Len() {
+			return false
+		}
```

Comment:
OK I think this check is not needed because the array length is part of its type, which is checked above. Please feel free to verify this with a test, though.